optaplanner-examples/src/main/java/org/optaplanner/examples/coachshuttlegathering/solver/CoachShuttleGatheringConstraintProvider.java

```diff
@@ -0,0 +1,142 @@
+package org.optaplanner.examples.coachshuttlegathering.solver;
+
+import java.util.function.Function;
+
+import org.optaplanner.core.api.score.buildin.hardsoftlong.HardSoftLongScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintCollectors;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.core.api.score.stream.Joiners;
+import org.optaplanner.examples.coachshuttlegathering.domain.Bus;
+import org.optaplanner.examples.coachshuttlegathering.domain.BusStop;
+import org.optaplanner.examples.coachshuttlegathering.domain.Coach;
+import org.optaplanner.examples.coachshuttlegathering.domain.Shuttle;
+import org.optaplanner.examples.coachshuttlegathering.domain.StopOrHub;
+
+public class CoachShuttleGatheringConstraintProvider implements ConstraintProvider {
+    final static String CONSTRAINT_PACKAGE = "org.optaplanner.examples.coachshuttlegathering.solver";
+
+    @Override
+    public Constraint[] defineConstraints(ConstraintFactory constraintFactory) {
+        return new Constraint[] {
+                coachStopLimit(constraintFactory),
+                shuttleCapacity(constraintFactory),
+                coachCapacity(constraintFactory),
+                coachCapacityShuttleButNoShuttle(constraintFactory),
+                coachCapacityCorrection(constraintFactory),
+                transportTime(constraintFactory),
+                shuttleDestinationIsCoachOrHub(constraintFactory),
+                shuttleSetupCost(constraintFactory),
+                distanceFromPrevious(constraintFactory),
+                distanceBusStopToBusDestination(constraintFactory),
+                distanceCoachDirectlyToDestination(constraintFactory)
+        };
+    }
+
+    Constraint coachStopLimit(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(Coach.class)
+                .join(BusStop.class, Joiners.equal(coach -> coach, BusStop::getBus))
+                .groupBy((coach, busStop) -> coach, ConstraintCollectors.countBi())
```

Comment:
Let's `import static ...ConstraintCollectors...`
AFAIK the other constraint providers do it too.